node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1480,14 +1485,24 @@ async fn fetch_finality_signatures_by_block_header(
 }
 
 /// Returns the EraId whose switch block should be used to obtain validator weights.
-fn get_era_id_for_validators_retrieval(era_id: &EraId) -> EraId {
-    if *era_id != EraId::from(0) {
-        // For eras > 0 we need to use validator set from the previous era.
+fn get_era_id_for_validators_retrieval(
+    era_id: &EraId,
+    last_emergency_restart: Option<EraId>,
+) -> EraId {
+    if *era_id != EraId::from(0) && last_emergency_restart != Some(*era_id) {
+        // For eras > 0 which are not the last emergency restart eras we need to use validator set
+        // from the previous era
```

Comment:
If there are multiple emergency restarts we need to do that for each of them, not just the last one.
Maybe add a TODO for that? Will probably be a bigger issue.

Reply:
Added in 05f4c9d7c547e81e0b8c2c681e569d095b907084